### PR TITLE
claim-schema changes per meeting

### PR DESCRIPTION
**Pull request by ryandgoulding**
Several changes to claim-schema.json were brought up during the team meeting
earlier.  As such, the changes were made, and a corresponding GoLang binding
was generated.  The generator was severely improved to allow remapping
properties at an arbitrary depth.  Hitherto, this was impossible, and would
consistently break the generator.

Additionally, it was discovered that the auto-generator wasn't automatically
attributing a license to the client binding.  This was changed, and is
configurable through generated-license-header.txt.

Tests were modified to ensure this is working properly.

Signed-off-by: Ryan Goulding <rgouldin@redhat.com>

### CODE REVIEW

File: claim-schema.json [hunk 1]
```diff
@@ -1,54 +1,69 @@
...
-              "description": "A description of a host using the Unix lshw command."
-            }
+          "additionalProperties": {
+            "type": "object",
+            "description": "A description of a host using the Unix lshw command."
           }
         },
-        "junitResults": {
+        "results": {
           "type": "object",
```
Comment (spenced) [inline claim-schema.json:52]: Isn't this an array? Or is the array of results a value in this object?

File: claim-schema.json [hunk 2]
```diff
@@ -1,54 +1,69 @@
 {
   "$schema": "http://json-schema.org/draft-07/schema#",
-  "version": "0.0.1",
+  "version": "v0.0.5",
```
[the remark this answers is not in the document]
Comment (ryandgoulding) [reply claim-schema.json:3]: It is intentional;  GoLang module semantic versioning prepends versions with a "v".

File: claim-schema.json [hunk 3]
```diff
@@ -1,54 +1,69 @@
...
   "properties": {
     "claim": {
-      "additionalProperties": false,
       "type": "object",
       "properties": {
         "startTime": {
           "type": "string",
           "format": "date-time",
-          "description": "The claim evaluation start time."
+          "description": "The claim evaluation UTC start time."
```
[the remark this answers is not in the document]
Comment (ryandgoulding) [reply claim-schema.json:13]: Sure.

File: claim-schema.json [hunk 4]
```diff
@@ -1,54 +1,69 @@
...
+  "version": "v0.0.5",
   "description": "Represents a test-network-function claim.",
-  "additionalProperties": false,
+  "type": "object",
   "properties": {
     "claim": {
-      "additionalProperties": false,
       "type": "object",
       "properties": {
         "startTime": {
```
[the remark this answers is not in the document]
Comment (ryandgoulding) [reply claim-schema.json:10]: Will do.

File: claim-schema.json (hunk 1 above)
Comment (ryandgoulding) [reply claim-schema.json:52]: The latter.

**Review by spenced — APPROVED** [review]

**Comment by ryandgoulding** [comment]
Further feedback will be incorporated as a part of CTONET-622

**Pull request merged**